### ISSUE TITLE
Proposal for parsing pagination metadata

### ISSUE THREAD

**Issue by rualatngua** [pedsmoreira/premiere#2]
Hi @pedsmoreira, I can't find a way to implement pagination. I usually, add more pagination data as metadata like this
```json
{
   "metadata": {
        "resultSet": {
             "totalItems": 300
        }
   },
   "response": [
          <array of data here>
   ]
}
```

So I suggest PremiereJS expose a hook so that we can parse metadata and store it in Store maybe. What do you think? Thank for all your effort!

**Comment by pedsmoreira** [comment]
Hi @rualatngua, thanks for the feedback. I think it's a great idea! Having the option to customize the response parsing would indeed be very helpful.

I'll keep this issue opened until it's implemented.
Thanks

**Comment by pedsmoreira** [comment]
Hi @rualatngua. This is being implemented on v0.2. There's a PR open with the version changes. 

The v0.2 should be released in the next days, once I fix the documentation.

**Comment by pedsmoreira** [comment]
The documentation is still to be improved, but it's now possible to provide a callback to intercept the response. When calling a store or model method, just provide a `callback` property in the `options`. This callback receives the response and should return the data to be processed.

Given your example, this is how it would look like:

```typescript
const callback = (response: AxiosResponse) => {
    const data = response.data;

    SomeVariableToStoreMetadata.store(data.metadata);
    return data.response;
};
User.index({callback});
```
